api/v1beta2/types.go

```diff
@@ -140,12 +142,186 @@ var (
 	ResourceTypeResourceGroup = ResourceType("resourceGroup")
 )
 
+// SecurityGroupRuleAction represents the actions for a Security Group Rule.
+// +kubebuilder:validation:Enum=allow;deny
+type SecurityGroupRuleAction string
+
+const (
+	// SecurityGroupRuleActionAllow defines that the Rule should allow traffic.
+	SecurityGroupRuleActionAllow SecurityGroupRuleAction = vpcv1.NetworkACLRuleActionAllowConst
+	// SecurityGroupRuleActionDeny defines that the Rule should deny traffic.
+	SecurityGroupRuleActionDeny SecurityGroupRuleAction = vpcv1.NetworkACLRuleActionDenyConst
+)
+
+// SecurityGroupRuleDirection represents the directions for a Security Group Rule.
+// +kubebuilder:validation:Enum=inbound;outbound
+type SecurityGroupRuleDirection string
+
+const (
+	// SecurityGroupRuleDirectionInbound defines the Rule is for inbound traffic.
+	SecurityGroupRuleDirectionInbound SecurityGroupRuleDirection = vpcv1.NetworkACLRuleDirectionInboundConst
+	// SecurityGroupRuleDirectionOutbound defines the Rule is for outbound traffic.
+	SecurityGroupRuleDirectionOutbound SecurityGroupRuleDirection = vpcv1.NetworkACLRuleDirectionOutboundConst
+)
+
+// SecurityGroupRuleProtocol represents the protocols for a Security Group Rule.
+// +kubebuilder:validation:Enum=all;icmp;tcp;udp
+type SecurityGroupRuleProtocol string
+
+const (
+	// SecurityGroupRuleProtocolAll defines the Rule is for all network protocols.
+	SecurityGroupRuleProtocolAll SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolAllConst
+	// SecurityGroupRuleProtocolICMP defiens the Rule is for ICMP network protocol.
+	SecurityGroupRuleProtocolICMP SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolIcmpConst
```

Comment:
```suggestion
	SecurityGroupRuleProtocolICMP SecurityGroupRuleProtocol = vpcv1.NetworkACLRuleProtocolICMPConst
```
lets maintain uniformity

Reply:
That is the name of the constant in the VPC code, I cannot change that
```
// Constants associated with the NetworkACLRule.Protocol property.
// The protocol to enforce.
const (
	NetworkACLRuleProtocolAllConst  = "all"
	NetworkACLRuleProtocolIcmpConst = "icmp"
	NetworkACLRuleProtocolTCPConst  = "tcp"
	NetworkACLRuleProtocolUDPConst  = "udp"
)
```

Reply:
If it is referred as `Icmp` in that package, lets do change in our code to follow same standard wherever possible.

Reply:
I think the standard is `ICMP` actually, which is also what the docs follow too. I think the SDK itself is inaccurate.
![image](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/assets/17167414/39c3955e-aac0-4e35-a3c8-33da91bf4ad6)

Reply:
I changed this one reference to `Icmp` to match the SDK constant name, have left remaining references as `ICMP`